File: remote-control/src/test/unit/com/thoughtworks/selenium/grid/remotecontrol/unregistration_request_test.rb
require File.expand_path(File.dirname(__FILE__) + "/../../../../../unit_test_helper")

include_class 'com.thoughtworks.selenium.grid.remotecontrol.UnregistrationRequest'

class UnregistrationRequestTest < Test::Unit::TestCase

  test "Post to Hub using URL provided in constructor for unregistration request" do
    request = UnregistrationRequest.new "http://thehub.url:4444", "", "", ""
    assert_equal "http://thehub.url:4444/registration-manager/unregister", request.post_method.get_uri.to_string
  end

end

File: remote-control/src/test/unit/java/com/thoughtworks/selenium/grid/remotecontrol/UnregistrationRequestTest.java
package com.thoughtworks.selenium.grid.remotecontrol;

import junit.framework.Assert;
import static junit.framework.Assert.assertEquals;
import org.jbehave.classmock.UsingClassMock;
import org.junit.Test;

import java.io.IOException;

/**
 * {@link com.thoughtworks.selenium.grid.remotecontrol.UnregistrationRequest} unit test class.
 */
public class UnregistrationRequestTest extends UsingClassMock {

    @Test
    public void postToHubUsingURLProvidedInConstructor() throws IOException {
      final HubRequest request = new UnregistrationRequest("http://thehub.url:4444", "", "", "");
        assertEquals("http://thehub.url:4444/registration-manager/unregister", request.postMethod().getURI().toString());
    }

}